go/libraries/doltcore/sqle/dprocedures/dolt_checkout.go

```diff
@@ -282,18 +375,33 @@ func checkoutNewBranch(ctx *sql.Context, dbName string, dbData env.DbData, apr *
 	sess := dsess.DSessFromSess(ctx.Session)
 	err = commitTransaction(ctx, sess, rsc)
 	if err != nil {
-		return err
+		return "", "", err
 	}
 
-	wsRef, err := ref.WorkingSetRefForHead(ref.NewBranchRef(newBranchName))
-	if err != nil {
-		return err
+	var remoteAndBranch string
+	if remoteName != "" {
+		remoteAndBranch = fmt.Sprintf("%s/%s", remoteName, remoteBranchName)
 	}
 
-	return sess.SwitchWorkingSet(ctx, dbName, wsRef)
+	if isGlobal {
+		return newBranchName, remoteAndBranch, doGlobalCheckout(ctx, newBranchName, apr.Contains(cli.ForceFlag))
+	} else {
+
+		wsRef, err := ref.WorkingSetRefForHead(ref.NewBranchRef(newBranchName))
+		if err != nil {
+			return "", "", err
+		}
+
+		err = sess.SwitchWorkingSet(ctx, dbName, wsRef)
+		if err != nil {
+			return "", "", err
+		}
+	}
+
+	return newBranchName, remoteAndBranch, nil
 }
 
-func checkoutBranch(ctx *sql.Context, dbName string, branchName string) error {
+func checkoutBranch(ctx *sql.Context, dbName string, branchName string, apr *argparser.ArgParseResults) error {
```

Comment:
In a file which also has `doDoltCheckout` and `doltCheckout`, this function blends in. What this function is actually doing is updating the branch reference, so say that.

Reply:
Since it goes alongside `checkoutNewBranch` and `checkoutRemoteBranch`, I renamed it to `checkoutExistingBranch` and added a docstring.